contracts/TalentLayerEscrow.sol

```diff
@@ -1007,18 +1102,21 @@ contract TalentLayerEscrow is
      * @param _amount The core escrow amount
      * @param _originServiceFeeRate the %fee (per ten thousands) asked by the platform for each service created on the platform
      * @param _originValidatedProposalFeeRate the %fee (per ten thousands) asked by the platform for each validates service on the platform
+     * @param _referralAmount the lump sum optional amount to be sent to the referrer, if any
      * @return totalEscrowAmount The total amount to be paid by the buyer (including all fees + escrow) The amount to transfer
      */
     function _calculateTotalWithFees(
         uint256 _amount,
         uint16 _originServiceFeeRate,
-        uint16 _originValidatedProposalFeeRate
+        uint16 _originValidatedProposalFeeRate,
+        uint256 _referralAmount
     ) private view returns (uint256 totalEscrowAmount) {
-        return
+        totalEscrowAmount =
             _amount +
```

Comment:
I just realize there is an issue here. The system can only work if we are taking the same fees that we do today on the _amount, for the _referralAmount. With fees it becomes sybil resistant
So here, _originServiceFeeRate, _originValidatedProposalFeeRate and _originValidatedProposalFeeRate should be applied to _referralAmount